js/ai/src/generate.ts

```diff
@@ -355,86 +355,93 @@ export interface GenerateStreamResponse<O extends z.ZodTypeAny = z.ZodTypeAny> {
   get response(): Promise<GenerateResponse<O>>;
 }
 
-function createPromise<T>(): {
-  resolve: (result: T) => unknown;
-  reject: (err: unknown) => unknown;
+// A Task is a common pattern for creating a handle to both an awaitable and the ability to trigger it.
+// first popularized by C#.
+interface Task<T> {
+  resolve: (result: T) => void;
+  reject: (err: unknown) => void;
   promise: Promise<T>;
-} {
-  let resolve, reject;
+}
+
+function createTask<T>(): Task<T> {
+  let resolve: unknown, reject: unknown;
   let promise = new Promise<T>((res, rej) => ([resolve, reject] = [res, rej]));
-  return { resolve, reject, promise };
+  return {
+    resolve: resolve as Task<T>['resolve'],
+    reject: reject as Task<T>['reject'],
+    promise,
+  };
+}
+
+class Channel<T> implements AsyncIterable<T> {
```

Comment:
nit/optional: this is neat and can probably be reused in other places... maybe move it to core?

Reply:
Will move this and Task to Core.